plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -216,8 +236,13 @@ protected PasswordAuthentication getPasswordAuthentication() {
      * <a href="https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/migrationGuides/V8_V12.md#miscellaneous">migration guide</a> for mode details:
      */
     private BlobServiceClientBuilder applyLocationMode(final BlobServiceClientBuilder builder, final AzureStorageSettings settings) {
-        final StorageConnectionString storageConnectionString = StorageConnectionString.create(settings.getConnectString(), logger);
-        final StorageEndpoint endpoint = storageConnectionString.getBlobEndpoint();
+        final StorageEndpoint endpoint;
+        if (settings.useManagedIdentityCredential()) {
+            endpoint = getStorageBlobEndpoint(settings);
```

Comment:
Any thoughts on consolidating this into `StorageConnectionString.create()` so we can do this everywhere where `StorageConnectionString.create()` is used?

Reply:
Or to put it in another way - what is the difference between "getStorageBlobEndpoint" and `getBlobEndpoint()` under StorageConnectionString?

Reply:
StorageConnectionString.create() is a method provided in `com.azure.storage.common.implementation.connectionstring.StorageConnectionString#create` which do not have the flexibility to make changes.

In short, `StorageConnectionString.create()` accepts a connection string to generate the blob endpoints. With managed identity, there is no connection string, hence we cannot use this method to generate the blob endpoints.

Therefore, `getStorageBlobEndpoint` is actually derived from StorageConnectionString.create() such that it will generate the blob endpoints without a connection string.